contracts/interchain-token-service/tests/deploy_remote_canonical_token.rs

```diff
@@ -62,9 +64,42 @@ fn deploy_remote_canonical_token_succeeds() {
     }
     .abi_encode(&env);
 
-    let deployed_token_id = client
-        .mock_all_auths_allowing_non_root_auth()
-        .deploy_remote_canonical_token(&token_address, &destination_chain, &spender, &gas_token);
+    let transfer_token_auth = build_mock_invoke!(
```

Comment:
Use build_mock_auth here as well perhaps and then pass in the `.invoke` field below for sub invokes